dashboard/app/models/contact_rollups_pardot_memory.rb

```diff
@@ -145,11 +158,19 @@ def self.update_pardot_prospects
         old_prospect_data,
         new_contact_data
       )
-      save_sync_results(submissions, errors, Time.now.utc) if submissions.present?
+
+      if submissions.present? && !dry_run
+        save_sync_results(submissions, errors, Time.now.utc)
+      end
```

Comment:
nit: slightly shorter
```suggestion
      save_sync_results(submissions, errors, Time.now.utc) if submissions.present? && !dry_run
```

Reply:
I did what @molly-moen suggested and moved this logic into the top of the `save_sync_results` method. LMK if you object.